data-service/src/main/java/com/amazonaws/blox/dataservice/model/Cluster.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.model;
+
+import java.util.StringJoiner;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+
+@Data
+@Builder
+// required for builder
+@AllArgsConstructor
+// required for mapstruct
+@NoArgsConstructor
+public class Cluster {
+  private static final String UNDERSCORE = "_";
```

Comment:
Good catch, but I don't think this disallow customer to use underscore in their cluster/environment name. We only use the underscore to form the hashkey in DDB, and it seems we do not need to get the cluster/environment name/id from the hashkey.

Reply:
I think the actual place that this underscore will matter is in the EnvironmentId object? Because we do try to decompose the accountID and cluster from the key. We may think about the impact there. https://github.com/blox/blox/blob/dev/data-service/src/main/java/com/amazonaws/blox/dataservice/model/EnvironmentId.java#L46

Reply:
Oh you are right. We should reconsider this.

Reply:
Agreed. I'd say we should use the same separators as ARNs, so identifiers that are valid in ARNs will also be valid in our DB key.

Reply:
So we will use cluster name not cluster arn across all of our apis and ddb tables?